x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/optimizer/rules/logical/FoldNullTests.java

```diff
@@ -34,11 +144,23 @@ public void testNullFoldingIsNull() {
     public void testGenericNullableExpression() {
         FoldNull rule = new FoldNull();
         // arithmetic
-        assertNullLiteral(rule.rule(new Add(EMPTY, getFieldAttribute(), NULL)));
+        assertNullLiteral(rule.rule(new Add(EMPTY, getFieldAttribute("a"), NULL)));
         // comparison
-        assertNullLiteral(rule.rule(greaterThanOf(getFieldAttribute(), NULL)));
+        assertNullLiteral(rule.rule(greaterThanOf(getFieldAttribute("a"), NULL)));
         // regex
         assertNullLiteral(rule.rule(new RLike(EMPTY, NULL, new RLikePattern("123"))));
+        // date functions
```

Comment:
The testGenericNullableExpression variant from LogicalPlanOptimizerTests was more extensive, thus the additional lines here.